dlt/destinations/impl/clickhouse/clickhouse.py

```diff
@@ -0,0 +1,416 @@
+                    file_path,
+                    "Unsigned Azure Blob Storage access from ClickHouse isn't supported as yet.",
+                )
+
+            # Authenticated access.
+            account_name = staging_credentials.azure_storage_account_name
+            storage_account_url = (
+                f"https://{staging_credentials.azure_storage_account_name}.blob.core.windows.net"
+            )
+            account_key = staging_credentials.azure_storage_account_key
+            container_name = bucket_url.netloc
+            blobpath = bucket_url.path
+
+            table_function = (
+                "SELECT * FROM"
+                f" azureBlobStorage('{storage_account_url}','{container_name}','{blobpath}','{account_name}','{account_key}','{clickhouse_format}','{compression}')"
+            )
+            statement = f"INSERT INTO {qualified_table_name} {table_function}"
+        elif not bucket_path:
+            # Local filesystem.
+            if file_extension == "parquet":
+                compression = "auto"
+            else:
+                compression = "gz" if FileStorage.is_gzipped(file_path) else "none"
+            try:
+                with clickhouse_connect.create_client(
+                    host=client.credentials.host,
+                    port=client.credentials.http_port,
+                    database=client.credentials.database,
+                    user_name=client.credentials.username,
+                    password=client.credentials.password,
+                    secure=bool(client.credentials.secure),
+                ) as clickhouse_connect_client:
+                    insert_file(
+                        clickhouse_connect_client,
+                        qualified_table_name,
+                        file_path,
+                        fmt=clickhouse_format,
+                        settings={
+                            "allow_experimental_lightweight_delete": 1,
+                            "allow_experimental_object_type": 1,
+                        },
+                        compression=compression,
+                    )
+            except clickhouse_connect.driver.exceptions.Error as e:
+                raise LoadJobTerminalException(
+                    file_path,
+                    f"ClickHouse connection failed due to {e}.",
+                ) from e
+        else:
+            raise LoadJobTerminalException(
+                file_path,
+                f"ClickHouse loader does not support '{bucket_scheme}' filesystem.",
+            )
+
+        # Don't use dbapi driver for local files.
+        if bucket_path:
+            with client.begin_transaction():
+                client.execute_sql(statement)
+
+    def state(self) -> TLoadJobState:
+        return "completed"
+
+    def exception(self) -> str:
+        raise NotImplementedError()
+
+
+class ClickHouseMergeJob(SqlMergeJob):
+    @classmethod
+    def _to_temp_table(cls, select_sql: str, temp_table_name: str) -> str:
```

Comment:
this is still failing. some problems with table aliasing
```
DB::Exception: Missing columns: 'd.node_id' 'd.url' 'd.id' while processing query: 'SELECT 1 FROM dlt.github_3_202404091123570088_staging___issues AS s WHERE (d.id = id) OR ((d.url = url) AND (d.node_id = node_id))', required columns: 'id' 'node_id' 'url' 'd.id' 'd.url' 'd.node_id', maybe you meant: 'id', 'node_id' or 'url': While processing 1 IN ((SELECT 1 FROM (SELECT 1 FROM dlt.github_3_202404091123570088_staging___issues AS s WHERE (d.id = s.id) OR ((d.url = s.url) AND (d.node_id = s.node_id))) LIMIT 1) AS _subquery5).
```

I expect this will be the biggest problem to fix